src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { deleteTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  isTemp?: boolean;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  inputRef: React.RefObject<HTMLInputElement>;
+  todosToDelete: number[];
+  setTodosToDelete: React.Dispatch<React.SetStateAction<number[]>>;
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  isTemp = false,
+  setTodos,
+  setErrorMessage,
+  inputRef,
+  todosToDelete,
+  setTodosToDelete,
+}) => {
+  const { title, id, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [changedTitle, setChangedTitle] = useState(title);
+  const [loader, setLoader] = useState(false);
+
+  const handleChangeOfTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setChangedTitle(event.target.value);
+  };
+
+  const handleDelete = () => {
+    setLoader(true);
+
+    deleteTodo(todo.id)
+      .then(() => {
+        setTodos(prev => prev.filter(item => item.id !== todo.id));
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+      })
+      .catch(() => setErrorMessage('Unable to delete a todo'));
+  };
+
+  const handleChangeSubmit = () => {};
+
+  const eventListenerKeyboard = () => {};
+
+  const switchCompleted = () => {};
+
+  useEffect(() => {
+    for (let i = 0; i < todosToDelete.length; i++) {
+      if (todo.id === todosToDelete[i]) {
+        handleDelete();
+        setTodosToDelete(prev => prev.splice(i, 1));
+      }
+    }
+  }, [todosToDelete]);
```

Comment:
pls fix this issue
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/108538921/3eea1f0e-27c9-4a37-a40b-48dc448880f8)